acceptance/internal/config.go

```diff
@@ -149,3 +152,56 @@ func DoLoadConfig(t *testing.T, path string) TestConfig {
 
 	return config
 }
+
+// This function takes EnvMatrix and expands into a slice of environment configurations.
+// Each environment configuration is a slice of env vars in standard Golang format.
+// For example,
+//
+//	input: {"KEY": ["A", "B"], "OTHER": ["VALUE"]}
+//
+// output: [["KEY=A", "OTHER=VALUE"], ["KEY=B", "OTHER=VALUE"]]
+//
+// If any entries is an empty list, that variable is dropped from the matrix before processing.
+func ExpandEnvMatrix(matrix map[string][]string) [][]string {
+	result := [][]string{{}}
+
+	if len(matrix) == 0 {
+		return result
+	}
+
+	// Filter out keys with empty value slices
+	filteredMatrix := make(map[string][]string)
```

Comment:
Why do we need to do this? Better to just fail the test run indicating a misconfiguration?